### PR TITLE
npmjs deployment with circleci

### PR DESCRIPTION
**Pull request by subdavis**
I'm tired of not having this :P

### CODE REVIEW

**Comment by subdavis** [comment]
I've tested all the individual parts of this and they seem to work.  

To publish to npm, you'd need to a) update package.json's version and merge to master and b) push a tag on that commit, then c) merge to master.

I'm pretty sure tags won't survive a squash-and-merge so we will have to probably dedicate PRs explicitly for release.  

Ready for review.

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@girder/components",
-  "version": "2.0.0",
+  "version": "2.0.1",
```
Comment (subdavis) [inline package.json]: I've already released this by hand while testing token auth, so it needs to be merged.

**Comment by subdavis** [comment]
This may not actually work.   For the publish job to run, the CI runner should be invoked by a tagged commit on master.  As far as I'm aware, that condition is impossible to satisfy.  Either squash-and-merge destroys your tag or the merge commit created by GitHub is the target of the CI job, which is 1 commit ahead of the commit you would have tagged.

**Comment by zachmullen** [comment]
Are we ok to just tag master after the version change gets merged?

**Comment by subdavis** [comment]
I honestly don't know.  I believed until now that simply pushing a tag wouldn't cause CI to run again for a commit that had already passed.

If that's wrong, then yes.

**Comment by zachmullen** [comment]
IIRC pushing a tag *does* trigger CI, I'm pretty sure that's how I do it for Girder.

**Comment by subdavis** [comment]
Then this is ready for merge and we can give it a try.

**Pull request merged**